Sources/Data Model/Audience/UserAttribute.swift

```diff
@@ -142,43 +142,43 @@ extension UserAttribute {
         case .lt:
             // user attribute "less than" this condition value
             // so evaluate if this condition value "isGreater" than the user attribute value
-            return try value!.isGreater(than: rawAttributeValue!)
+            return try value!.isGreater(than: rawAttributeValue!, condition: stringRepresentation, name: nameFinal)
         case .le:
             // user attribute "less than" or equal this condition value
             // so evaluate if this condition value "isGreater" than or equal the user attribute value
-            return try value!.isGreaterOrEqual(than: rawAttributeValue!)
+            return try value!.isGreaterOrEqual(than: rawAttributeValue!, condition: stringRepresentation, name: nameFinal)
         case .gt:
             // user attribute "greater than" this condition value
             // so evaluate if this condition value "isLess" than the user attribute value
-            return try value!.isLess(than: rawAttributeValue!)
+            return try value!.isLess(than: rawAttributeValue!, condition: stringRepresentation, name: nameFinal)
         case .ge:
             // user attribute "greater than or equal" this condition value
             // so evaluate if this condition value "isLess" than or equal the user attribute value
-            return try value!.isLessOrEqual(than: rawAttributeValue!)
+            return try value!.isLessOrEqual(than: rawAttributeValue!, condition: stringRepresentation, name: nameFinal)
         // semantic versioning seems unique.  the comarison is to compare verion but the passed in version is the target version.
         case .semver_eq:
             guard let targetValue = AttributeValue(value: rawAttributeValue) else {
-                throw OptimizelyError.evaluateAttributeInvalidType("attribute value \(nameFinal) invalid type")
+                throw OptimizelyError.evaluateAttributeInvalidCondition("attribute value \(nameFinal) invalid type")
```

Comment:
Don't we need sem-version type checking and log error cases? "invalid sem-ver format", etc?